holder.yml

```diff
@@ -56,60 +56,108 @@ paths:
+              anyOf:
+                -
+                  {
+                    "type": "object",
+                    "description": "Data necessary to initiate the exchange."
+                  }
+                -
+                  $ref: "#/components/schemas/NotifyPresentationAvailableRequest"
       responses:
         "200":
-          description: Proceed with presentation
+          description: Proceed with exchange.
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/NotifyPresentationAvailableResponse"
+                $ref: "#/components/schemas/VerifiablePresentationRequest"
         "400":
-          description: Request for presentation is malformed
+          description: Request is malformed.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
         "501":
-          description: Not implemented
+          description: Service not implemented.
         "500":
-          description: internal error
-  /presentations/submissions:
+          description: Internal server error.
+  /exchanges/{exchange-id}/{transaction-uuid}:
     post:
-      summary: Provide a presentation for a holder to store.
-      operationId: storePresentation
-      description: "Delivery endpoint for a client-holder to provide a presentation in compliance with the Verifiable Presentation Request provided in the Notification call. This server-holder MUST match the 'domain' and 'challenge' to a 'domain' and 'challenge' previously sent in a Notification Response, and not already received. The server-holder MUST verify the Presentation. The server MUST store the Presentation if verification passes."
+      summary: Receives information related to an existing exchange.
+      operationId: receiveExchangeData
+      description:
+        A client can use this endpoint to continue the exchange of information
+        associated with an initiated exchange. A Verifiable Presentation is
+        sent to this endpoint. If the server has received all of the information
+        it needs, a 200 OK is returned with either an empty response or a
+        Verifiable Presentation containing requested credentials by the client.
+        If the server requires more information, it returns a Verifiable
+        Presentation Request. A request that the server cannot understand
+        results in an error.
+      parameters:
+        - in: path
+          name: exchange-id
+          description:
+            A human readable exchange type identifier such as "diplomas".
+          required: true
+          schema:
+            type: string
+            pattern: "[a-z0-9\\-]{3,}"
+        - in: path
+          name: transaction-uuid
+          required: true
+          schema:
+            type: string
+            pattern: "([0-9a-f]{8}\b-[0-9a-f]{4}\b-[0-9a-f]{4}\b-[0-9a-f]{4}\b-[0-9a-f]{12}){0,1}"
```

Comment:
hmm might be better to make this an IRI same comment as before, it may show up in other JSON-LD `id` fields.

Reply:
Need a use case to understand what you're saying. I'm sure it's relevant, and yes, we need to be careful about things like this showing up in JSON-LD `id` fields... that said, it's a UUID in a URL, which can fit in a JSON-LD `id` field.

Reply:
A. `d2dfc100-9384-49e4-b547-5e7d84ec5bf5` -> not an IRI

B. `urn:uuid:d2dfc100-9384-49e4-b547-5e7d84ec5bf5` -> IRI

C. `https://app.example/d2dfc100-9384-49e4-b547-5e7d84ec5bf5` -> IRI

D. `https://app.example/urn:uuid:d2dfc100-9384-49e4-b547-5e7d84ec5bf5` -> IRI

My understanding of JSON-LD is that only B-D would be acceptable for a `credentialSubject.id` or `issuer.id`... does that match you understanding?

Reply:
> does that match you understanding?

Yes, that matches my understanding and I now understand what you are saying. 

Based on the discussion during the telecon yesterday, I believe the group decided that we need to be very careful about this point. That is, if a server does not have control over what an `id` or `name` can be, we should not allow it to be a part of a path parameter (to avoid ugly URL encoding). I believe we agreed to that as a general rule and will need to determine how this fact affects how future path parameters are defined and used.